src/hooks/useCommunityChart.ts

```diff
@@ -0,0 +1,72 @@
+// };
+import { useEffect, useState } from 'react';
+
+import { IDateStats } from '@/constants/affiliates';
+import { AffiliatesProgramMetric, AffiliatesProgramPeriod } from '@/constants/charts';
+
+import { log } from '@/lib/telemetry';
+
+import { useEndpointsConfig } from './useEndpointsConfig';
+
+export const useCommunityChart = (selectedChartMetric: AffiliatesProgramMetric) => {
+  const { affiliatesBaseUrl } = useEndpointsConfig();
+  // State to keep track of selected period
+  const [selectedPeriod, setSelectedPeriod] = useState<AffiliatesProgramPeriod>(
+    AffiliatesProgramPeriod.PeriodAllTime
+  );
+
+  // State to store the metric data for the currently selected period
+  const [metricData, setMetricData] = useState<{ date: number; cumulativeAmount: number }[]>([]);
+
+  useEffect(() => {
+    fetchMetricData();
+  }, [selectedChartMetric, selectedPeriod]);
+
+  const getStartDate = (): string => {
+    const currentTime = new Date();
+
+    switch (selectedPeriod) {
+      case AffiliatesProgramPeriod.Period1d:
+        return new Date(currentTime.setDate(currentTime.getDate() - 1)).toISOString();
+      case AffiliatesProgramPeriod.Period7d:
+        return new Date(currentTime.setDate(currentTime.getDate() - 7)).toISOString();
+      case AffiliatesProgramPeriod.Period30d:
+        return new Date(currentTime.setMonth(currentTime.getMonth() - 1)).toISOString();
+      case AffiliatesProgramPeriod.Period90d:
+        return new Date(currentTime.setMonth(currentTime.getMonth() - 3)).toISOString();
+      case AffiliatesProgramPeriod.PeriodAllTime:
+        return new Date(0).toISOString(); // The earliest possible date
+      default:
+        throw new Error('Invalid rolling window value');
```

Comment:
this shouldn't ever error if we are using `enums`. Maybe make the default in line with `PeriodAllTime`?